benches/mod.rs

```diff
@@ -59,6 +60,20 @@ fn bench_with_init<const N: usize>(b: &mut test::Bencher) {
     b.bytes = N as u64;
 }
 
+// Used to benchmark the benefit of `getrandom_uninit` compared to
+// zero-initializing a buffer and then using `getrandom` (`bench_with_init`
+// above).
+#[inline(always)]
+fn bench_uninit<const N: usize>(b: &mut test::Bencher) {
+    let mut ab = AlignedBuffer::<N>::new();
+    let buf = ab.buf();
+    // SAFETY: `buf` doesn't escape this scope.
+    let buf = unsafe { slice_as_uninit_mut(buf) };
+    b.iter(|| {
+        let _ = getrandom::getrandom_uninit_slice(buf);
+    })
+}
```

Comment:
What do the `bench_uninit` benchmarks give us over the `bench` benchmarks above (which also don't initialize the buffer?

Reply:
> What do the bench_uninit benchmarks give us over the bench benchmarks above (which also don't initialize the buffer?

Since the reason for adding `getrandom_uninit` is performance, AFAICT, it would be surprising to not have benchmarks for it that show it is faster in cases we think matter.

FWIW, I think the benchmarks would reflect real-world use much better if they took this shape instead of what they look like presently:

```rust
let mut buffer = [0u8; 32];
getrandom(&mut buffer)?;
```
In particular, `AlignedBuffer` seems to be over-optimizing compared to what we'd expect people to do IRL.

Reply:
I think I recommended it in a separate comment: it could be worth to adapt the benches from #279.

Reply:
I agree here, having two benchmarks types (like #279), where one looks like:
```rust
let mut buf = [0u8; N];
getrandom::getrandom(&mut buf[..]).unwrap();
test::black_box(&buf);
```
and another that looks like:
```rust
let mut uninit_buf = [MaybeUninit::<u8>::uninit(); N];
let buf = getrandom_uninit(&mut uninit_buf).unwrap();
test::black_box(&buf);
```
would be more useful than what we currently have and more closely resemble actual use-cases.